client/landing/stepper/declarative-flow/tailored-ecommerce-flow.ts

```diff
@@ -48,16 +48,16 @@ function getPlanFromRecurType( recurType: string ) {
 
 const ecommerceFlow: Flow = {
 	name: ECOMMERCE_FLOW,
-	useSteps() {
-		const recurType = useSelect(
+	isSignupFlow: true,
+	useSignupStartEventProps() {
```

Comment:
See https://github.com/Automattic/wp-calypso/pull/87758/files#r1500191298.
We could try calling `useSignupStartEventProps` outside the `useEffect` might we might need to add checks to prevent the event triggering multiple time.

For the sake of unblocking this PR, propose we don't implement `useSignupStartEventProps` for now, trigger the event as it previously did, and iterate in a follow-up PR. WDYT?

Reply:
@daledupreez @gabrielcaires  WDYT if we move `useSignupStartEventProps` as the following:

```
  const eventProps = flow.useSignupStartEventProps?.() ?? {};                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
  useEffect( () => {                                                                                                                                                                                                                                                                                           
    if ( flow.isSignupFlow && isFlowStart() ) {                                                                                                                                                                                                                                                                  
      recordSignupStart( flow.name, ref, eventProps );                                                                                                                                                                                                                                                     
    }                                                                                                                                                                                                                                                                                                    
  }, [ flow, ref ] );
```

Reply:
@taipeicoder I tried a couple different ways of moving `flow.useSignupStartEventProps` outside of the `useEffect` and didn't have much luck.

I'm leaning towards your suggestion of removing it for this PR and then iterating in a follow-up.

Reply:
Yep, sounds good!

Reply:
Should we also remove the function for this iteration, especially if we're not calling it?

Reply:
I've removed the function and the type in 5b604fec115b62601e6b795b41c9333ddb9bf927. That way we can name things however we like in the follow-up PR.